typescript/nomad-deploy/src/incremental/index.ts

```diff
@@ -0,0 +1,60 @@
+import * as ethers from 'ethers';
+
+import { NomadContext } from '@nomad-xyz/sdk/src';
+
+function toCall(t: ethers.ethers.PopulatedTransaction) {
+  return { to: t.to!, data: ethers.utils.toUtf8Bytes(t.data!) };
+}
+
+export async function enrollSpoke(
+  sdk: NomadContext,
+  spokeDomain: number,
+  watchers: string[],
+) {
+  let hubCore = await sdk.governorCore();
+  let hubBridge = sdk.mustGetBridge(hubCore.domain);
+
+  let spokeCore = await sdk.mustGetCore(spokeDomain);
+  let spokeBridge = await sdk.mustGetBridge(spokeDomain);
+  let batch = await hubCore.newGovernanceBatch();
+
+  // enroll watchers
+  await Promise.all(
+    watchers.map(async (watcher) => {
+      const call =
+        await hubCore.xAppConnectionManager.populateTransaction.watcherPermission(
+          watcher,
+          spokeDomain,
+        );
+      batch.pushLocal(toCall(call));
+    }),
+  );
+
+  // enroll replica
+  const newReplicaAtOldChainAddress = hubCore.getReplica(spokeDomain)?.address;
```

Comment:
might be clearer to name it `hubReplicaOfSpoke`

Reply:
+1